src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidUnreliableStreamRead.Fixer.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    /// <summary>
+    /// CA2022: <inheritdoc cref="AvoidUnreliableStreamReadTitle"/>
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class AvoidUnreliableStreamReadFixer : CodeFixProvider
+    {
+        private const string Async = nameof(Async);
+        private const string ReadExactly = nameof(ReadExactly);
+        private const string ReadExactlyAsync = nameof(ReadExactlyAsync);
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(AvoidUnreliableStreamReadAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetRequiredSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span, getInnermostNodeForTie: true);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            var semanticModel = await context.Document.GetRequiredSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var operation = semanticModel.GetOperation(node, context.CancellationToken);
+
+            if (operation is not IInvocationOperation invocation ||
+                invocation.Instance is null)
+            {
+                return;
+            }
+
+            var compilation = semanticModel.Compilation;
+            var streamType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemIOStream);
+
+            if (streamType is null)
+            {
+                return;
+            }
+
+            var isAsyncInvocation = invocation.TargetMethod.Name.EndsWith(Async, StringComparison.Ordinal);
+            var readExactlyMethods = streamType.GetMembers(isAsyncInvocation ? ReadExactlyAsync : ReadExactly)
+                .OfType<IMethodSymbol>()
+                .ToImmutableArray();
```

Comment:
NIT: The ReadExactly and ReadExactlyAsync methods added simultaneously, doesn't really need to be filtered conditionally, just loading one of them should be enough

Reply:
Thanks, simplified the check in ed5e7a8003899a9344e9492ae4c06b14bbcaa5f0.